### PR TITLE
Feature/#50 add edit relation dialog step2

### LINKED ISSUE #50: Add / Edit relation dialog - step 2
We need to add dropdowns to allow the user select the key field on each table.

It should be displayed as a dropdown, BUT:
  - When the user click to open the dropdown, the box is open...
  -  Showing the field hierarchy.
  - The fields marked as object (have children) cannot be selected.
  - The children field can be selected (the layout should display the hierarchy like we do in the table).
  - [This could be implemented in a separate issue]Once a user chooses a field it should be displayed in the combo (showing the full path, something like address>country>_id

I'm open to proposal, and it would be a good idea to think wether to implement this from scratch or search if there's some component already implemented

Mocks:

image

image

image

### PR DESCRIPTION
**Pull request by deletidev**
Fixes #50

### CODE REVIEW

File: src/common/components/table-pk-picker/components/field-tree.business.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import { Tick } from '../../icons/tick-icon.component';
-import { OptionVm } from '../table-pk-picker.model';
+import { PkOptionVm } from '../table-pk-picker.model';
 import classes from '../table-pk-picker.component.module.css';
 
 export const generateOptions = (
```
Comment (brauliodiez) [inline src/common/components/table-pk-picker/components/field-tree.business.tsx:5]: Add reference to this issue (unit tests):

https://github.com/Lemoncode/mongo-modeler/issues/116

File: src/pods/edit-realtion/edit-relation.business.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
-import { DatabaseSchemaVm } from '@/core/providers/canvas-schema';
+import { DropdownOptionVm, PkOptionVm } from '@/common/components';
+import { GUID } from '@/core/model';
+import {
+  DatabaseSchemaVm,
+  FieldVm,
+  TableVm,
+} from '@/core/providers/canvas-schema';
 
 export const mapRelationsTipeToDropdonwVm = (): DropdownOptionVm[] => [
```
Comment (brauliodiez) [inline src/pods/edit-realtion/edit-relation.business.ts:9]: Mention this issue (unit tests)

https://github.com/Lemoncode/mongo-modeler/issues/118

File: src/common/components/table-pk-picker/components/field-tree.business.tsx (hunk 1 above)
Comment (deletidev) [reply src/common/components/table-pk-picker/components/field-tree.business.tsx:8]: It's fixed

File: src/pods/edit-realtion/edit-relation.business.ts (hunk 2 above)
Comment (deletidev) [reply src/pods/edit-realtion/edit-relation.business.ts:9]: It's fixed

**Pull request merged**